src/service/ampdoc-impl.js

```diff
@@ -418,6 +418,33 @@ export class AmpDoc {
     return v == null ? null : v;
   }
 
+  /**
+   * Returns map of an ampdoc's meta name value to content value
+   * @return {!Object<string, string>}
+   */
+  getMeta() {
+    const metaMap = map();
```

Comment:
Can we cache this?

Reply:
Commit 555d6685878941dd975f139963abd9ba55f6d3b9 takes a stab at this. Let me know if you agree with the implementation.